examples/desktop/misc/lorenz_animation.py

```diff
@@ -0,0 +1,92 @@
+"""
+Lorenz System Animation
+=======================
+
+Example the Lorenz attractor.
+"""
+
+# test_example = false
+# sphinx_gallery_pygfx_docs = 'animate 10s'
+
+import fastplotlib as fpl
+import numpy as np
+
+
+# generate data
+def lorenz(xyz, *, s=10, r=28, b=2.667):
+    """
+    Parameters
+    ----------
+    xyz : array-like, shape (3,)
+       Point of interest in three-dimensional space.
+    s, r, b : float
+       Parameters defining the Lorenz attractor.
+
+    Returns
+    -------
+    xyz_dot : array, shape (3,)
+       Values of the Lorenz attractor's partial derivatives at *xyz*.
+    """
+    x, y, z = xyz
+    x_dot = s * (y - x)
+    y_dot = r * x - y - x * z
+    z_dot = x * y - b * z
+    return np.array([x_dot, y_dot, z_dot])
+
+
+dt = 0.01
+num_steps = 3_000
+
+lorenz_data = list()
```

Comment:
preallocate an array to account for the 3,000 timesteps instead of appending to a list?